### PR TITLE
Server Auth API Implementation

### PR DESCRIPTION
**Pull request by NilsG-S**
This PR begins the transition from Firebase to the server API. It also includes some minor visual fixes, quality of life additions (React Hot Loader), and back end changes.

### CODE REVIEW

File: src/redux/actions.js [hunk 1]
```diff
@@ -55,29 +52,30 @@ export var student = {
...
-export var guest = {
-  role: AuthStates.GUEST
+export const guest = {
+  role: AuthStates.GUEST,
 };
 
 // action creators
 
-export const updateUser = (user) => {
+export function updateUser(user) {
```
Comment (asclines) [inline src/redux/actions.js]: What is the reasoning for switching to this function style?

File: src/routes/Settings/routes/Profile/components/Profile.jsx [hunk 2]
```diff
@@ -27,74 +25,66 @@ class Profile extends React.Component {
...
   }
 
   handleSubmit(event) {
     // This prevents a '?' from being appended to the URL
     event.preventDefault();
-    var data = {};
 
-    if(this.props.firstName !== this.state.firstName) {
-      data.firstName = this.state.firstName;
+    if (this.props.name !== this.state.name) {
```
Comment (asclines) [inline src/routes/Settings/routes/Profile/components/Profile.jsx]: What is the purpose of this block of code?

File: test/components/Auth/test-auth.jsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { shallow } from 'enzyme';
 
 import Auth from 'components/Auth/Auth.jsx';
 
-export default describe('Auth.jsx', () => {
+describe('Auth.jsx', () => {
```
Comment (asclines) [inline test/components/Auth/test-auth.jsx]: Why the removal of `export`?

Comment (NilsG-S) [inline test/components/Auth/test-auth.jsx]: As it turns out the export is not strictly necessary, so I decided to remove it. There shouldn't be a problem leaving the others in, I just felt that this was a bit cleaner.

File: src/routes/Settings/routes/Profile/components/Profile.jsx (hunk 2 above)
Comment (NilsG-S) [inline src/routes/Settings/routes/Profile/components/Profile.jsx]: Just a small optimization. I don't call the function unless the name has changed (`this.props.name` comes from redux, `this.state.name` is used to store local changes), that way I'm not hitting the server with useless requests.

File: src/redux/actions.js (hunk 1 above)
Comment (NilsG-S) [inline src/redux/actions.js]: This is more consistent with what I've used elsewhere. I typically use the `function` syntax for named functions and the arrow function syntax for anonymous functions. I can change it back if it doesn't mesh with the rest of the style we're using.

**Pull request merged**